src/App.tsx

```diff
@@ -26,57 +26,108 @@ type ReorderOptions = {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
 export const App: React.FC = () => {
+  const initialOptions: ReorderOptions = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  const [reorderOptions, setReorderOptions] =
+    useState<ReorderOptions>(initialOptions);
+
+  const handleSortAlphabetically = () => {
+    setReorderOptions(prevOptions => ({
+      ...prevOptions,
+      sortType: SortType.ALPHABET,
+    }));
+  };
+
+  const handleSortByLength = () => {
+    setReorderOptions(prevOptions => ({
+      ...prevOptions,
+      sortType: SortType.LENGTH,
+    }));
+  };
+
+  const handleReverse = () => {
+    setReorderOptions(prevOptions => ({
+      ...prevOptions,
+      isReversed: !prevOptions.isReversed,
+    }));
+  };
+
+  const handleReset = () => {
+    setReorderOptions(initialOptions);
+  };
+
+  const visibleGoods = getReorderedGoods(goodsFromServer, reorderOptions);
+
+  const isGoodsInOriginalOrder =
+    JSON.stringify(reorderOptions) === JSON.stringify(initialOptions);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={`button is-info ${reorderOptions.sortType === SortType.ALPHABET ? '' : 'is-light'}`}
+          onClick={handleSortAlphabetically}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={`button is-success ${reorderOptions.sortType === SortType.LENGTH ? '' : 'is-light'}`}
+          onClick={handleSortByLength}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={`button is-warning ${reorderOptions.isReversed ? '' : 'is-light'}`}
+          onClick={handleReverse}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {isGoodsInOriginalOrder ? null : (
+          <button
+            type="button"
+            className="button is-danger"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map((good, index) => (
+          <li key={index} data-cy="Good">
```

Comment:
don't use array index as a key prop
https://legacy.reactjs.org/docs/lists-and-keys.html#keys